linkml/generators/pydanticgen.py

```diff
@@ -547,6 +579,7 @@ def default_value_for_type(self, typ: str) -> str:
 
 @shared_arguments(PydanticGenerator)
 @click.option("--template_file", help="Optional jinja2 template to use for class generation")
+@click.option("--pydantic_version", help="Pydantic version to use (1 or 2)", default="1")
```

Comment:
Check out https://click.palletsprojects.com/en/8.1.x/options/#choice-options and then you should be able to remove the bespoke value checking in lines 597-600. I believe using that will also show the valid options in the `--help` output.